### ISSUE TITLE
Segmentation runtime

### ISSUE THREAD

**Issue by afilt** [MICS-Lab/thunder#13]
Hi @PierreMarza,

I'm running a segmentation task on `segpath_epithelial` with pre-computed embeddings:

```
Running dataset: segpath_epithelial
2026-04-13 23:06:52,459 - INFO -
🚀  Experiment Info
------------------------------------------------------------
Task   : segmentation
Dataset: segpath_epithelial
Model  : my_custom_vit_base_model
------------------------------------------------------------

Hyper-parameters
------------------------------------------------------------
type: frozen
batch_size: 64
num_workers: 40
lr: [0.001, 0.0001, 1e-05]
weight_decay: [0.0, 0.001, 0.0001]
epochs: 200
------------------------------------------------------------
2026-04-13 23:09:00,420 - INFO - Not found already trained best model. Training a model.
```

on 1 V100 GPUs (32 Gb) and 40 CPUs and have this terrible ETA:

```
  4%|▎         | 7/200 [9:28:22<260:53:27, 4866.36s/it]
```

How can I speed things up ? Thanks !

**Comment by afilt** [comment]
Okay I've just checked the paper:

```
Semantic segmentation — hyperparameters are summarized in Table S4. We perform different
numbers of epochs depending on the size of the datasets: 200 epochs for ocelot and pannuke,
respectively 9 and 21 for segp-ep and segp-ly as those two datasets are much larger. We leverage the
Segmenter [67] decoder as our segmentation probe.
```

Maybe worth to mention in the documentation ?

**Comment by PierreMarza** [comment]
Hi!

Yes, indeed, you should keep all thunder default hyperparameters for all tasks and datasets, except for the `segpath_epithelial` and `segpath_lymphocytes` datasets (`segmentation` task) for which you must override the number of epochs to `9` and `21` respectively (because of the size of the datasets).

I just added this information to the doc (https://mics-lab.github.io/thunder/guidelines/), thanks! Closing this issue accordingly (feel free to re-open if you have any other question/feedback).

**Comment by afilt** [comment]
Thanks for updating the documentation.
Additional note: I'm running into OOM errors and it seems related to saving the masks using `wandb` because the inference is running smoothly:

```python
    if task_type == "segmentation":
        mask_list = wb_mask(
            test_viz_im.permute((0, 2, 3, 1)).cpu().numpy(),
            test_viz_pred.cpu().numpy(),
            test_viz_gt.cpu().numpy(),
            cfg.dataset.classes,
        )
        wandb.log(
            {
                f"{wandb_base_folder}/test_predictions_lr_{best_ckpt_dict['lr']}_weight_decay_"
                f"{best_ckpt_dict['weight_decay']}": mask_list
            }
        )
```

**Comment by PierreMarza** [comment]
Thanks for reporting this. To the best of my knowledge, we have never experienced such an issue. Would you have more information to provide about your setting? For instance, how much RAM do you have access to?

**Comment by afilt** [comment]
Hi, I was running on 40 CPUs and 190 Go RAM. I'm working on a cluster that seems to prevently kill some processes that make the RAM increases suddenly (from past experience).

**Comment by PierreMarza** [comment]
Ok, I see! And if you remove the `wandb` mask saving code lines you shared, you don't have any OOM error anymore?

**Comment by afilt** [comment]
Yes absolutely. I don't have logs to share unfortunately except "Killed due to OOO error"; but using breakpoints I isolated the masks saving as being the culprit for the `segpath` datasets at least.
